### ISSUE TITLE
UI automation support for drag-and-drop

### ISSUE THREAD

**Issue by willibrandon** [dotnet/winforms#10705]
### .NET version

9

### Did it work in .NET Framework?

No

### Issue description

UI automation defines two control patterns for supporting drag-and-drop scenarios that expose information assistive technology can use to help an accessibility user complete a drag-and-drop operation.
https://learn.microsoft.com/en-us/windows/win32/winauto/ui-automation-support-for-drag-and-drop

I would like to ask whether there is any interest in implementing the [Drag](https://learn.microsoft.com/en-us/windows/win32/winauto/uiauto-implementingdrag) and [DropTarget](https://learn.microsoft.com/en-us/windows/win32/winauto/uiauto-implementingdroptarget) control patterns in effort to make drag-and-drop more accessible.

![AIWin_DragPattern_DropTargetPattern](https://github.com/dotnet/winforms/assets/5017479/e9d61913-bfa6-481e-94bf-a2485107f432)

### Steps to reproduce

Use AIWin Live Inspect to see that Drag and DropTarget control pattern properties do not exist.

**Comment by merriemcgaw** [comment]
This is such a cool idea, but it's not on our current radar for accessibility. These are not required patterns to our knowledge, and we recommend applications support keyboard actions when it comes to things like moving contents from one container to another. Note though that voice access may be using UIA so it's a cool thing to expose for that scenario.

**Comment by willibrandon** [comment]
> Note though that voice access may be using UIA so it's a cool thing to expose for that scenario.

Yes I agree, and that would be the exact goal here. Similar to what Narrator does now when you drag and rearrange a taskbar item on Windows 11 and it announces that as:

"Grab Developer Command Prompt for Visual Studio 2022 Preview. One running window. Drag complete."

I could be wrong, but I believe for something like this to work with a Windows Forms application, the items being dragged and the drop targets must first support the drag-and-drop related properties in such a way that they can be made accessible. Screen readers could then implement support for announcing the progress of a drag-and-drop operation by utilizing the associated `IUIAutomationDragPattern` and `IUIAutomationDropTargetPattern` client interfaces to access the drag-and drop accessibility information for the UI elements involved in the operation.

I understand if this isn't currently on your accessibility radar, but if there is any interest in pursuing this endeavor, I am here to help.
